### DISCUSSION TITLE
How about adding a page history to the wiki?

### DISCUSSION THREAD

**Discussion by nibreh** [datenstrom/community#196] [Ask a question] [answered]
It could be nice to create one day, a historical feature to save a copy of edited page, like a wiki.

I know [we can use dropbox](https://github.com/datenstrom/yellow/issues/7#issuecomment-51331194), but I don't use dropbox for [security & ethical reasons](https://en.wikipedia.org/wiki/Dropbox_%28service%29#Reception)

**What do you think about ?**

PS: By the way, is there a way to forbid users delete page ?

**Comment by nibreh** [comment]
Yes ! And there is a trash folder for deleted pages now. It's very nice and useful.

I suggested a historical feature for collaborative Wiki. I think about it but if there is a historical feature, when one user edit a page, this page must be locked. This is more work to do !

If I was good enough in hacking code, I would do it. We must find a developer to help Mark and ease the job :)

**Comment by nibreh** [comment]
> when one user edit a page, this page must be locked

Sorry for my mistake, but this is already in place. It says `Page has been modified by someone else` if users attempts to save a page at the same time.

**Comment by nibreh** [comment]
So we have:

- Trash folder to save deleted page
- Users creation account online, approved by the owner
- Users restriction to protect folders
- Locked pages to protect from overwriting

We miss : 

- Each time you save a page, a copy it's saved in a history folder.

**Comment by annaesvensson** [comment, marked as answer]
At the moment no developer is working on this, but would be lovely to have something like this in the future. Maybe someone wants to [make an extension](https://github.com/annaesvensson/yellow-publish) or improve the [wiki extension](https://github.com/annaesvensson/yellow-wiki) with a history feature? All that is needed is someone to volunteer. 😄
